### PR TITLE
use exporter-toolkit to add TLS and Basic Auth

### PR DESCRIPTION
**Pull request by fschlich**
This is mostly copied from SNMP and Redfish exporters, and requires a go-kit based logger. README.md mentions "Rework logging" as TODO, but doesn't mention any specifics - I personally wouldn't need levels for example...

### CODE REVIEW

**Comment by RichiH** [comment]
Also tagging @SuperQ for potential pair of eyes as it's taking things from snmp exporter.

File: go.mod [hunk 1]
```diff
@@ -4,21 +4,14 @@ go 1.11
...
-	github.com/prometheus/client_golang v0.9.2
-	github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90 // indirect
-	github.com/prometheus/common v0.3.0
-	github.com/prometheus/procfs v0.0.0-20190412120340-e22ddced7142 // indirect
-	github.com/sirupsen/logrus v1.4.1 // indirect
-	github.com/stretchr/testify v1.3.0 // indirect
+	github.com/hashicorp/go-multierror v1.0.0
+	github.com/prometheus/client_golang v1.7.1
+	github.com/prometheus/common v0.15.0
+	github.com/prometheus/exporter-toolkit v0.5.0
```
Comment (SuperQ) [inline go.mod:13]: This is a very old version, you should probably update this to the latest. We have simplified the interface and flags. See https://github.com/prometheus/node_exporter/pull/2393.

**Review by SuperQ — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (SuperQ) [reply go.mod:13]: `--telemetry-listen-address` will be replaced by the exporter toolkit `--web.listen-address`. The `--modbus-listen-address` will stay the same. But should probably be renamed to match the usual Prometheus conventions.

**Comment by DaAwesomeP** [comment]
@fschlich any updates on this pull?

**Comment by DaAwesomeP** [comment]
@fschlich Sounds good! I will create a new pull request with the latest exporter-toolkit and modbus_exporter branches.  I will potentially cherry pick your changes.

**Pull request closed without merging**